code/modules/mob/mob_defines.dm

```diff
@@ -106,6 +106,7 @@
 	var/candrop = 1
 	var/lastpuke = 0
 	var/unacidable = 0
+	var/nospells = 0 //Can't use spells. At all.
```

Comment:
That already exists, but as silencing is a variable on the spell, you have to HAVE the spell before it can be silenced
A helper proc to silence spells acquired after it was called would have to set silenced on each spell as you acquired it, which would require either a variable or an event listener, both of which you complained about